lib/radix.c

```diff
@@ -98,7 +98,7 @@ static char *rn_zeros, *rn_ones;
 #define rm_leaf rm_rmu.rmu_leaf /* extra field would make 32 bytes */
 
 /* Helper macros */
-#define squid_Bcmp(a, b, l) (l == 0 ? 0 : memcmp((caddr_t)(a), (caddr_t)(b), (u_long)l))
+#define squid_Bcmp(a, b, l) (l == 0 ? 0 : memcmp((char *)(a), (char *)(b), (u_long)l))
```

Comment:
This badly written macro is unused[^1]. It is best to delete it instead of adjusting its _problematic_ argument casts.

```suggestion
```

[^1]: I did not check carefully, but it looks like it is unused since inception in 2001 commit 06cdab8 or, in its original `Bcmp()` spelling, in 1997 commit 76f729d.